handlers/spendMsgHandler.go

```diff
@@ -0,0 +1,74 @@
+package handlers
+
+import (
+	"crypto/sha256"
+	"github.com/FourthState/plasma-mvp-sidechain/msgs"
+	"github.com/FourthState/plasma-mvp-sidechain/plasma"
+	"github.com/FourthState/plasma-mvp-sidechain/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"math/big"
+)
+
+// returns the next tx index in the current block
+type NextTxIndex func() uint16
+
+func NewSpendHandler(utxoStore store.UTXOStore, nextTxIndex NextTxIndex) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+		spendMsg, ok := msg.(msgs.SpendMsg)
+		if !ok {
+			panic("Msg does not implement SpendMsg")
+		}
+
+		txIndex := nextTxIndex()
+		blockHeight := big.NewInt(ctx.BlockHeight())
+
+		// construct the confirmation hash
+		merkleHash := spendMsg.MerkleHash()
+		header := ctx.BlockHeader().DataHash
```

Comment:
TODO: Double-check that DataHash doesn't change across two consecutive tx's being handled in same block